### ISSUE TITLE
[clang] `-Wcharacter-conversion` should warn on `wchar_t` conversions

### ISSUE THREAD

**Issue by eisenwave** [llvm/llvm-project#168854]
https://compiler-explorer.com/z/6zb99Y8Yj
```cpp
void f(char8_t c8, char16_t c16, wchar_t cw) {
    bool a = c8 == c16;
    bool b = c8 == cw;
}
```
Currently, only the first line raises `-Wcharacter-conversion`; the second line doesn't.

However, a comparison between `char8_t` and `wchar_t` is just as nonsensical as a comparison between `char8_t` and `char16_t`, so this seems like a false negative. That is especially true when considering that `wchar_t` is often UTF-16 or UTF-32, depending on the platform.

Also note that when seeing the the corresponding paper [P3695R2](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2025/p3695r2.html), SG16 Unicode unanimously voted to deprecate conversions between `char8_t` and `wchar_t`. We have since flipped on that decision because it is hypothetically possible to have a UTF-8 `wchar_t` in C2y, but the compiler doesn't really need to care about such far-fetched hypotheticals; it has platform knowledge.

CC @cor3ntin
